pkg/ingester/circuitbreaker.go

```diff
@@ -253,3 +255,61 @@ func (cb *circuitBreaker) recordResult(errs ...error) error {
 	cb.metrics.circuitBreakerResults.WithLabelValues(circuitBreakerResultSuccess).Inc()
 	return nil
 }
+
+type ingesterCircuitBreaker struct {
+	push *circuitBreaker
+	read *circuitBreaker
+}
+
+func newIngesterCircuitBreaker(pushCfg CircuitBreakerConfig, readCfg CircuitBreakerConfig, logger log.Logger, registerer prometheus.Registerer) *ingesterCircuitBreaker {
+	return &ingesterCircuitBreaker{
+		push: newCircuitBreaker(pushCfg, registerer, circuitBreakerPushRequestType, logger),
+		read: newCircuitBreaker(readCfg, registerer, circuitBreakerReadRequestType, logger),
+	}
+}
+
+func (cb *ingesterCircuitBreaker) activate() {
+	if cb == nil {
```

Comment:
do we need this check? I don't see how `cb` can be `nil` here.

Reply:
Correct